torchaudio/csrc/CMakeLists.txt

```diff
@@ -126,6 +126,51 @@ define_library(
   "${LIBTORCHAUDIO_COMPILE_DEFINITIONS}"
   )
 
+
+################################################################################
+# libtorchaudio_decoder.so
+################################################################################
+if (BUILD_CTC_DECODER)
+  set(
+    LIBTORCHAUDIO_DECODER_SOURCES
+    decoder/src/decoder/Decoder.h
+    decoder/src/decoder/LexiconDecoder.cpp
+    decoder/src/decoder/LexiconDecoder.h
+    decoder/src/decoder/Trie.cpp
+    decoder/src/decoder/Trie.h
+    decoder/src/decoder/Utils.cpp
+    decoder/src/decoder/Utils.h
+    decoder/src/decoder/lm/KenLM.cpp
+    decoder/src/decoder/lm/KenLM.h
+    decoder/src/decoder/lm/LM.h
+    decoder/src/dictionary/Defines.h
+    decoder/src/dictionary/Dictionary.cpp
+    decoder/src/dictionary/Dictionary.h
+    decoder/src/dictionary/String.cpp
+    decoder/src/dictionary/String.h
+    decoder/src/dictionary/System.cpp
+    decoder/src/dictionary/System.h
+    decoder/src/dictionary/Utils.cpp
+    decoder/src/dictionary/Utils.h
+    )
+  set(
+    LIBTORCHAUDIO_DECODER_DEFINITIONS
+    BUILD_CTC_DECODER
+    )
+  set(
+    LIBTORCHAUDIO_DECODER_DEPS
+    libtorchaudio
+    kenlm)
+  define_library(
+    libtorchaudio_decoder
+    "${LIBTORCHAUDIO_DECODER_SOURCES}"
+    "${PROJECT_SOURCE_DIR}"
+    "${LIBTORCHAUDIO_DECODER_DEPS}"
+    "${LIBTORCHAUDIO_COMPILE_DEFINITIONS};${LIBTORCHAUDIO_DECODER_DEFINITIONS}"
+    )
+endif()
+
+# TODO: Add libtorchaudio_decoder
```

Comment:
what is necessary for adding libtorchaudio_decoder?

Reply:
The definition of `TORCHAUDIO_LIBRARY` that is expected to be used from C++ integration via CMake. To make the decoder available in C++, we need to add `libtorchaudio_decoder` to it. However, the current way is not entirely correct, so we need to revise it.